File: app/src/main/java/com/github/gfx/helium/di/FragmentModule.java
package com.github.gfx.helium.di;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class FragmentModule {

    final Fragment fragment;

    public FragmentModule(Fragment fragment) {
        this.fragment = fragment;
    }

    @Named("fragment")
    @Provides
    public Context context() {
        return fragment.getContext();
    }

    @Provides
    public FragmentManager provideFragmentManager() {
        return fragment.getFragmentManager();
    }
}

File: app/src/main/java/com/github/gfx/helium/di/FragmentModule.kt
package com.github.gfx.helium.di

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager

import dagger.Module
import dagger.Provides

@Module
class FragmentModule(internal val fragment: Fragment) {

    @Provides
    fun context(): Context {
        return fragment.context
    }

    @Provides
    fun provideFragmentManager(): FragmentManager {
        return fragment.fragmentManager
    }
}
